### PR TITLE
[GEOT-6211] Add a command line tool to migrate NetCDF H2 indexes to a centralized database

### PR DESCRIPTION
**Pull request by aaime**

### CODE REVIEW

File: modules/plugin/coverage-multidim/netcdf/src/main/java/org/geotools/coverage/io/netcdf/tools/H2Migrate.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2018, Open Source Geospatial Foundation (OSGeo)
```
Comment (dromagnoli) [inline modules/plugin/coverage-multidim/netcdf/src/main/java/org/geotools/coverage/io/netcdf/tools/H2Migrate.java]: Happy new year :)
... 2019 :P

File: modules/plugin/coverage-multidim/netcdf/src/main/java/org/geotools/coverage/io/netcdf/tools/H2MigrateConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,190 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2018, Open Source Geospatial Foundation (OSGeo)
```
Comment (dromagnoli) [inline modules/plugin/coverage-multidim/netcdf/src/main/java/org/geotools/coverage/io/netcdf/tools/H2MigrateConfiguration.java]: 2019?

File: modules/plugin/coverage-multidim/netcdf/src/main/java/org/geotools/coverage/io/netcdf/tools/H2MigrateConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+        Properties properties = loadStoreProperties(configPath);
+
+        this.sourceStoreConfiguration = properties;
+    }
+
+    public Properties loadStoreProperties(String configPath) {
+        File configFile = new File(configPath);
+        if (!configFile.exists() || !configFile.isFile()) {
+            throw new ConfigurationException(
+                    "Target store configuration file does not exist" + configPath);
```
Comment (dromagnoli) [inline modules/plugin/coverage-multidim/netcdf/src/main/java/org/geotools/coverage/io/netcdf/tools/H2MigrateConfiguration.java]: I would add an empty space after the "exist" word to better read the configPath

File: modules/plugin/coverage-multidim/netcdf/src/main/java/org/geotools/coverage/io/netcdf/tools/H2Migrator.java [hunk 4]
```diff
@@ -0,0 +1,369 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2018, Open Source Geospatial Foundation (OSGeo)
```
Comment (dromagnoli) [inline modules/plugin/coverage-multidim/netcdf/src/main/java/org/geotools/coverage/io/netcdf/tools/H2Migrator.java]: ?2019?

File: modules/plugin/coverage-multidim/netcdf/src/main/java/org/geotools/coverage/io/netcdf/tools/H2Migrator.java [hunk 5]
```diff
@@ -0,0 +1,369 @@
...
+                                                h2Writer)));
+            }
+            for (Future future : futures) {
+                try {
+                    future.get();
+                } catch (Exception e) {
+                    if (configuration.isFailureIgnored()) {
+                        LOGGER.log(Level.WARNING, "Failed to migrate file ", e);
+                    } else {
+                        throw new MigrationException("Failed to migrate file", e);
```
Comment (dromagnoli) [inline modules/plugin/coverage-multidim/netcdf/src/main/java/org/geotools/coverage/io/netcdf/tools/H2Migrator.java]: I would add an emptySpace after file: ".. migrate file "

File: modules/plugin/coverage-multidim/netcdf/src/main/java/org/geotools/coverage/io/netcdf/tools/LocationFeatureCollection.java [hunk 6]
```diff
@@ -0,0 +1,70 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2018, Open Source Geospatial Foundation (OSGeo)
```
Comment (dromagnoli) [inline modules/plugin/coverage-multidim/netcdf/src/main/java/org/geotools/coverage/io/netcdf/tools/LocationFeatureCollection.java]: 2019?

File: modules/plugin/coverage-multidim/netcdf/src/main/java/org/geotools/coverage/io/netcdf/tools/LogWriter.java [hunk 7]
```diff
@@ -0,0 +1,46 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2018, Open Source Geospatial Foundation (OSGeo)
```
Comment (dromagnoli) [inline modules/plugin/coverage-multidim/netcdf/src/main/java/org/geotools/coverage/io/netcdf/tools/LogWriter.java]: 2019?

File: modules/plugin/coverage-multidim/netcdf/src/main/java/org/geotools/coverage/io/netcdf/tools/MigrationException.java [hunk 8]
```diff
@@ -0,0 +1,28 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2018, Open Source Geospatial Foundation (OSGeo)
```
Comment (dromagnoli) [inline modules/plugin/coverage-multidim/netcdf/src/main/java/org/geotools/coverage/io/netcdf/tools/MigrationException.java]: 2019?

File: modules/plugin/coverage-multidim/netcdf/src/test/java/org/geotools/coverage/io/netcdf/tools/H2MigratorTest.java [hunk 9]
```diff
@@ -0,0 +1,378 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2018, Open Source Geospatial Foundation (OSGeo)
```
Comment (dromagnoli) [inline modules/plugin/coverage-multidim/netcdf/src/test/java/org/geotools/coverage/io/netcdf/tools/H2MigratorTest.java]: 2019?

Comment (ianturton) [reply modules/plugin/coverage-multidim/netcdf/src/test/java/org/geotools/coverage/io/netcdf/tools/H2MigratorTest.java]: :-)

**Comment by aaime** [comment]
@dromagnoli made a few changes and made the build pass on Windows, and added documentation page in GeoTools linking to the GeoServer one, and over there, usage instructions.
Could you have a look?

See also: https://github.com/geoserver/geoserver/pull/3325

**Pull request merged**